### ISSUE TITLE
"last" isn't getting set to true

### ISSUE THREAD

**Issue by alexwhitman** [mariano/node-db-mysql#25]
``` javascript
db.query.select('foo').from('bar')
    .on('each', function(row, index, last) {
        console.log(index + ': ' + last);
    }).execute();
```

If 'bar' contains 3 rows I'd expect the output to be:

```
0: false
1: false
2: true
```

but I'm instead getting:

```
0: false
1: false
2: false
```

I'm using 0.6.7 with MySQL.

**Comment by alexwhitman** [comment]
Looks like the cause is because of the

```
iterator != end
```

check at https://github.com/mariano/node-db/blob/master/query.cc#L752 which means that

```
iterator == end
```

at https://github.com/mariano/node-db/blob/master/query.cc#L759 will always be false.

**Comment by alexwhitman** [comment]
I've fixed this in node-db and sent a pull request https://github.com/mariano/node-db/pull/1

**Comment by mariano** [comment]
Fixed in 5e9e21da63152fb94486908592eb82537bb5c23a

Thanks!
